osf/models/preprint.py

```diff
@@ -1054,6 +1054,9 @@ def update_has_data_links(self, auth: Auth, has_data_links: bool, log: bool = Tr
         if self.has_data_links == has_data_links:
             return
 
+        if has_data_links == 'no':
+            self.data_links = []
```

Comment:
So the thing I don't like about this is that it throws away data from the API request silently. If you do send a request in with `has_data_links = 'no'` but you include data links, the data links are deleted and the user doesn't know why.  I think it's better to catch these situations in the serializer the way they were before, just not to catch the situations when the data (like data-links) is falsey.

I think it's generally good to clear out the data links if has_data_links is 'no', but it's not ideal to ignore a malformed API request. So as long as it's caught properly in the serializer and errored, this code can probably stay. That would allow people to set the `has_data_links` to 'no' without setting anything else in the API request, then it would clear out the data links properly. Does this all make sense?